cmd/clairctl/export.go

```diff
@@ -22,18 +25,31 @@ var ExportCmd = &cli.Command{
 	Name:      "export-updaters",
 	Action:    exportAction,
 	Usage:     "run updaters and export results",
-	ArgsUsage: "[out]",
+	ArgsUsage: "[out[.gz|.zst]]",
 	Flags: []cli.Flag{
 		// Strict can be used to check that updaters still work.
 		&cli.BoolFlag{
 			Name:  "strict",
 			Usage: "Return non-zero exit when updaters report errors.",
 		},
+		&cli.BoolFlag{
+			Name:    "gzip",
+			Aliases: []string{"g"},
+			Usage:   "Compress output with gzip.",
+		},
+		&cli.BoolFlag{
+			Name:    "zstd",
+			Aliases: []string{"z"},
+			Usage:   "Compress output with zstd.",
+		},
 	},
 	Description: `Run configured exporters and export to a file.
 
-   A configuration file is needed to run this command, see 'clairctl help'
-   for how to specify one.`, // NB this has spaces, not tabs.
+If a file name is supplied and ends with ".gz" or ".zst", output will be
+compressed with gzip or zstd compression, respectively.
+
```

Comment:
Maybe mention how the file extension overrides the boolean flags regards to compression?

Reply:
Made it work so that the flags override the filename.